pkg/upgrade/machine_deployment.go

```diff
@@ -43,25 +40,29 @@ func (u *MachineDeploymentUpgrader) Upgrade() error {
 	return u.upgradeMachineDeployments(machineDeployments)
 }
 
-func (u *MachineDeploymentUpgrader) listMachineDeployments() (*clusterapiv1alpha1.MachineDeploymentList, error) {
+func (u *MachineDeploymentUpgrader) listMachineDeployments() (*clusterapiv1alpha2.MachineDeploymentList, error) {
 	u.log.Info("Listing machine deployments")
 
-	listOptions := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("cluster.k8s.io/cluster-name=%s,set=node", u.clusterName),
+	selectors := []ctrlclient.ListOption{
+		ctrlclient.MatchingLabels{
+			"cluster.x-k8s.io/cluster-name": u.clusterName,
+		},
+		ctrlclient.InNamespace(u.clusterNamespace),
 	}
-
-	machineDeployments, err := u.managementClusterAPIClient.MachineDeployments(u.clusterNamespace).List(listOptions)
+	list := &clusterapiv1alpha2.MachineDeploymentList{}
+	err := u.ctrlClient.List(context.TODO(), list, selectors...)
 	if err != nil {
 		return nil, errors.Wrap(err, "error listing machines")
 	}
 
-	return machineDeployments, nil
+	return list, nil
 }
 
-func (u *MachineDeploymentUpgrader) upgradeMachineDeployments(list *clusterapiv1alpha1.MachineDeploymentList) error {
+func (u *MachineDeploymentUpgrader) upgradeMachineDeployments(list *clusterapiv1alpha2.MachineDeploymentList) error {
 	for _, machineDeployment := range list.Items {
 		// Skip any machineDeployments that already have this upgrade annotation id
-		if val, ok := machineDeployment.Spec.Template.Annotations[UpgradeIDAnnotationKey]; ok && val == u.upgradeID {
+		if val := machineDeployment.Spec.Template.Spec.Version; val != nil && *val == u.upgradeID {
+			// if val, ok := machineDeployment.Spec.Template.Annotations[UpgradeIDAnnotationKey]; ok && val == u.upgradeID {
```

Comment:
delete code instead of commenting it, or add a comment explaining why it's being commented out instead of deleted

Reply:
Can't delete it. Need to undo change on L64